spec/mainApiHandler.spec.ts

```diff
@@ -0,0 +1,304 @@
+import { activityDetection } from '../src/app/activity-detection';
+import '../src/app/main-api-handler';
+import { protocolHandler } from '../src/app/protocol-handler';
+import { screenSnippet } from '../src/app/screen-snippet-handler';
+import * as windowActions from '../src/app/window-actions';
+import { windowHandler } from '../src/app/window-handler';
+import * as utils from '../src/app/window-utils';
+import { apiCmds, apiName } from '../src/common/api-interface';
+import { logger } from '../src/common/logger';
+import { BrowserWindow, ipcMain } from './__mocks__/electron';
+
+jest.mock('electron-log');
+
+jest.mock('../src/app/protocol-handler', () => {
+    return {
+        protocolHandler: {
+            setPreloadWebContents: jest.fn(),
+        },
+    };
+});
+
+jest.mock('../src/app/screen-snippet-handler', () => {
+    return {
+        screenSnippet: {
+            capture: jest.fn(),
+        },
+    };
+});
+
+jest.mock('../src/app/window-actions', () => {
+    return {
+        activate: jest.fn(),
+        handleKeyPress: jest.fn(),
+    };
+});
+
+jest.mock('../src/app/window-handler', () => {
+    return {
+        windowHandler: {
+            closeWindow: jest.fn(),
+            createNotificationSettingsWindow: jest.fn(),
+            createScreenPickerWindow: jest.fn(),
+            createScreenSharingIndicatorWindow: jest.fn(),
+            isOnline: false,
+        },
+    };
+});
+
+jest.mock('../src/app/window-utils', () => {
+    return {
+        downloadManagerAction: jest.fn(),
+        isValidWindow: jest.fn(() => true),
+        sanitize: jest.fn(),
+        setDataUrl: jest.fn(),
+        showBadgeCount: jest.fn(),
+        showPopupMenu: jest.fn(),
+        updateLocale: jest.fn(),
+    };
+});
+
+jest.mock('../src/common/logger', () => {
+    return {
+        logger: {
+            setLoggerWindow: jest.fn(),
+        },
+    };
+});
+
+jest.mock('../src/app/config-handler', () => {
+    return {
+        config: {
+            getConfigFields: jest.fn(() => true),
+        },
+    };
+});
+
+jest.mock('../src/app/activity-detection', () => {
+    return {
+        activityDetection: {
+            setWindowAndThreshold: jest.fn(),
+        },
+    };
+});
+
+jest.mock('../src/common/i18n');
+
+describe('main api handler', () => {
+
+    beforeEach(() => {
+        jest.clearAllMocks();
+    });
+
+    describe('symphony-api events', () => {
```

Comment:
@vrbsm Also need to include negative test cases for `!args` and `inValidWindow`

Reply:
:+1: